### ISSUE TITLE
Implement changes to Android native code

### ISSUE THREAD

**Issue by yaronyg** [thaliproject/Thali_CordovaPlugin#894]
You need to coordinate with @artemjackson so that when you have these changes ready he will have already changed the node code to expect these changes and have tests for them. It's likely to be a while so don't expect to work on this for a bit.

Please note that the list below is NOT a substitute for reading and analyzing the changes in thaliMobileNative.js yourself and making sure you have everything covered.
- [x] If you get a multiConnect or disconnect you need to return the right error - 'Platform does not support connect'
- [x] The connect method will only contain the Bluetooth MAC and not the generation for a peer but this isn't a problem since that is all we need to connect
- [x] We now have an error (and therefore need a test) for a scenario where we try to open a TCP/IP listener but cannot because there are no available ports
- [x] peerAvailabilityChanged now breaks out the peerIdentifier (read: the bluetooth address) and the generation (the one byte) into two seperate fields.

**Comment by evabishchevich** [comment]
Do we need native test for 3rd point?

**Comment by evabishchevich** [comment]
Implemented 3 of 4 features. Need node code to test fixes

**Comment by yaronyg** [comment]
Yup, you are blocked on https://zube.io/projects/3188/tickets/1367. Once it is done you will have the node tests you need.

**Comment by evabishchevich** [comment]
We have to inform that there is no available port in case of getting 
java.net.SocketException: socket failed: EMFILE (Too many open files)

**Comment by yaronyg** [comment]
One more hour and one more test run and then we get a PR!

**Comment by evabishchevich** [comment]
It took a little bit more time.  I don't think that I have to send PR before I merge it with  node code. Am I right?

**Comment by yaronyg** [comment]
Please don't submit this PR until you have the node code that you need to test against.

**Comment by yaronyg** [comment]
@evabishchevich Please make sure to close this bug when you get the PR accepted.

**Comment by evabishchevich** [comment]
@yaronyg I will close that after #1368 is merged

**Comment by evabishchevich** [comment]
@yaronyg  I'm going to close this issue when ios_yarong_merge is merged into ios branch

**Comment by evabishchevich** [comment]
@yaronyg thaliproject/Thali_CordovaPlugin#1568 is merged into our "super" ios
